### PR TITLE
Add utilities to generate EnvVars for secret

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

This PR adds utilities for generating EnvVars for a secret. The EnvVars will be set for repository wrappers to provide credentials.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kopia/command/storage/secret_utils.go [hunk 1]
```diff
@@ -56,3 +68,123 @@ func checkSkipSSLVerifyFromMap(m map[string]string) bool {
...
+		envVars = append(envVars, getEnvVarWithSecretRef(aws.SessionToken, s.Name, secrets.AWSSessionToken))
+	}
+	return envVars, nil
+}
+
+func getEnvSpecForAzureCredentialSecret(s *v1.Secret) ([]v1.EnvVar, error) {
+	envVars := []v1.EnvVar{}
+	envVars = append(
+		envVars,
+		getEnvVarWithSecretRef("AZURE_STORAGE_ACCOUNT", s.Name, secrets.AzureStorageAccountID),
```
Comment (akankshakumari393) [inline pkg/kopia/command/storage/secret_utils.go:119]: can we declare "AZURE_STORAGE_ACCOUNT", "AZURE_STORAGE_KEY",  "AZURE_STORAGE_DOMAIN" as constant?

File: pkg/kopia/command/storage/secret_utils.go [hunk 2]
```diff
@@ -57,6 +74,94 @@ func locationType(m map[string][]byte) LocType {
...
+	if s == nil {
+		return nil, errors.New("Secret cannot be nil")
+	}
+	secType := string(s.Type)
+	switch secType {
+	case secrets.AWSSecretType:
+		return getEnvSpecForAWSCredentialSecret(s)
+	case secrets.AzureSecretType:
+		return getEnvSpecForAzureCredentialSecret(s)
+	}
```
Comment (shlokc9) [inline pkg/kopia/command/storage/secret_utils.go:89]: What about location secrets of type GCS and NFS stores?

Comment (ankitjain235) [reply pkg/kopia/command/storage/secret_utils.go:89]: For GCS, we would use the creds file and we don't have required ENV available.
For NFS, we don't need the credentials.

**Review by pavannd1 — COMMENTED** [review]
Looks good overall. We can make required changes for web identity token if required and then merge this

**Review by pavannd1 — APPROVED** [review]

**Pull request merged**